tests/rust-apps-wasm/rust-sdk/wit-sdk/miden.wit

```diff
@@ -138,6 +138,13 @@ interface core-types {
 
 }
 
+
+/// Note script interface that is expected to be implemented by the note script.
+interface note-script {
+    note-script: func();
```

Comment:
Is this what we want to call the user-facing note script entrypoint? I would consider using `main`, so that writing note scripts feels mostly like writing an executable, or at least from a user perspective it is more apparent that `main` is being called by the system (Miden runtime), rather than from their own code. Not sure if `rustc` complains about that without `#![no_main]`, but  I think that would make note scripts feel a bit more intuitive.

Is there a convention for how these are named in Miden currently? We could just go with whatever that is too, but looking for ways to make things feel more intuitive/natural for users (to the extent we can).

Reply:
> Is this what we want to call the user-facing note script entrypoint? 

Yes. The names are drafts and are not fixed.

> I would consider using `main`, so that writing note scripts feels mostly like writing an executable, or at least from a user perspective it is more apparent that `main` is being called by the system (Miden runtime), rather than from their own code. Not sure if `rustc` complains about that without `#![no_main]`, but I think that would make note scripts feel a bit more intuitive.

I've tried to build the Wasm command component (`main` entrypoint) before, but the amount of `wasi:*` code it brought in was quite high. 
I revisited this approach, and it appears that all the `wasi:*` code could be removed by crafting a custom adapter. I made an [#350](https://github.com/0xPolygonMiden/compiler/issues/350) issue for this, but it would take some time. So I suggest we merge this PR as is and take care of it later.

> Is there a convention for how these are named in Miden currently? 

No. We can come up with one.